enterprise/coderd/provisionerkeys_test.go

```diff
@@ -0,0 +1,108 @@
+	"github.com/coder/coder/v2/enterprise/coderd/coderdenttest"
+	"github.com/coder/coder/v2/enterprise/coderd/license"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestProvisionerKeys(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong*10)
+	t.Cleanup(cancel)
+	dv := coderdtest.DeploymentValues(t)
+	dv.Experiments = []string{string(codersdk.ExperimentMultiOrganization)}
+	client, owner := coderdenttest.New(t, &coderdenttest.Options{
+		Options: &coderdtest.Options{
+			DeploymentValues: dv,
+		},
+		LicenseOptions: &coderdenttest.LicenseOptions{
+			Features: license.Features{
+				codersdk.FeatureMultipleOrganizations: 1,
+			},
+		},
+	})
+	orgAdmin, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID, rbac.ScopedRoleOrgAdmin(owner.OrganizationID))
+	member, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID)
+	otherOrg := coderdtest.CreateOrganization(t, client, coderdtest.CreateOrganizationOptions{})
+	outsideOrgAdmin, _ := coderdtest.CreateAnotherUser(t, client, otherOrg.ID, rbac.ScopedRoleOrgAdmin(otherOrg.ID))
+
+	// member cannot create a provisioner key
+	_, err := member.CreateProvisionerKey(ctx, otherOrg.ID, codersdk.CreateProvisionerKeyRequest{
+		Name: "key",
+	})
+	require.ErrorContains(t, err, "Resource not found")
+
+	// member cannot list provisioner keys
+	_, err = member.ListProvisionerKeys(ctx, otherOrg.ID)
+	require.ErrorContains(t, err, "Resource not found")
+
+	// member cannot delete a provisioner key
+	err = member.DeleteProvisionerKey(ctx, otherOrg.ID, "key")
+	require.ErrorContains(t, err, "Resource not found")
+
+	// outside org admin cannot create a provisioner key
+	_, err = outsideOrgAdmin.CreateProvisionerKey(ctx, owner.OrganizationID, codersdk.CreateProvisionerKeyRequest{
+		Name: "key",
+	})
+	require.ErrorContains(t, err, "Resource not found")
+
+	// outside org admin cannot list provisioner keys
+	_, err = outsideOrgAdmin.ListProvisionerKeys(ctx, owner.OrganizationID)
+	require.ErrorContains(t, err, "Resource not found")
+
+	// outside org admin cannot delete a provisioner key
+	err = outsideOrgAdmin.DeleteProvisionerKey(ctx, owner.OrganizationID, "key")
+	require.ErrorContains(t, err, "Resource not found")
+
+	// org admin can list provisioner keys and get an empty list
+	keys, err := orgAdmin.ListProvisionerKeys(ctx, owner.OrganizationID)
+	require.NoError(t, err, "org admin list provisioner keys")
+	require.Len(t, keys, 0, "org admin list provisioner keys")
+
+	// org admin can create a provisioner key
+	_, err = orgAdmin.CreateProvisionerKey(ctx, owner.OrganizationID, codersdk.CreateProvisionerKeyRequest{
+		Name: "Key", // case insensitive
+	})
+	require.NoError(t, err, "org admin create provisioner key")
+
+	// org admin can conflict on name creating a provisioner key
+	_, err = orgAdmin.CreateProvisionerKey(ctx, owner.OrganizationID, codersdk.CreateProvisionerKeyRequest{
+		Name: "key",
```

Comment:
Maybe change the capitalization of this to like `KEY` to really check the case insensitivity. This being lowercase could accidentally pass.